src/functions.jl

```diff
@@ -492,4 +492,37 @@ function LinearAlgebra.vecdot(x::AbstractArray{Variable}, y::AbstractArray{Linea
     vecdot!(zero(QuadraticFunction{T}), x, y)
 end
 
+# vcat
+function _vcat!(dest::AbstractVector{<:AffineFunction},
+                i::Integer)
+    @boundscheck i == lastindex(dest) + 1 || throw(DimensionMismatch())
+    dest
+end
+
+function _vcat!(dest::AbstractVector{<:AffineFunction},
+                i::Integer,
+                source::AbstractVector{<:AffineFunction},
+                remaining::Vararg{<:AbstractVector{<:AffineFunction}, N}) where {N}
+    @boundscheck i >= firstindex(dest) && (i + length(source) - 1) <= lastindex(dest) || throw(DimensionMismatch())
+    @inbounds for s in source
+        copyto!(dest[i], s)
+        i += 1
+    end
+    _vcat!(dest, i, remaining...)
+end
+
+function vcat!(y::AbstractVector{<:AffineFunction},
+               args::Vararg{<:AbstractVector{<:AffineFunction}, N}) where N
+    @inbounds for i in eachindex(y)
+        if isassigned(y, i)
+            zero!(y[i])
+        else
+            y[i] = zero(AffineFunction{T})
```

Comment:
`T` not defined. But let's just get rid of the `!isassigned(y, i)` case here and in other locations (not in this PR); I don't think that code path is getting hit anymore.

Reply:
Oof, yeah. Good catch. Right, this would only get triggered if you used `similar()` to construct the destination, and we'll hit an undefined reference error if it happens in the future.

Reply:
Actually we do hit that code path here: https://github.com/tkoolen/SimpleQP.jl/blob/86e67cacf535213e14df327e0fddec208050abaf/test/functions.jl#L100 but only because of the test itself. I'll update the test.